### DISCUSSION TITLE
Run event with different parameters via Chain Reaction

### DISCUSSION THREAD

**Discussion by kranskydog** [jhuckaby/Cronicle#661] [General]
In the Run Event API, it is possible to provide custom parameters for a specific job, so I can create an event and pass different parameters each time I run that event.

I want to be able to do the same thing via Chain Reaction - that is, pass parameters to the chained event, so I don't have to create a different event for the same thing with different parameters. I know I can pass data between the chained events, but that is not quite the same.

Is it possible to pass a complete Event Data Object via Chain Reaction the same way it can be done via the API?

**Comment by jhuckaby** [comment]
Not yet, but I **love** that idea and I am going to code it up right now.  Should be released later tonight.

**Comment by jhuckaby** [comment]
Released [Cronicle v0.9.38](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.38) with this new requested feature implemented.

Docs: https://github.com/jhuckaby/Cronicle/blob/master/docs/Plugins.md#chain-params

**Comment by kranskydog** [comment]
Thanks. I'll give a try

**Comment by kranskydog** [comment]
Not sure what I may be doing wrong, but I can seem to get it to work

I have a plugin that defines 2 parameters PARAM1 and PARAM2

I have an event1 which does

#!/usr/bin/env bash
echo $PARAM1
echo $PARAM2
echo '{ "chain": "elo3mvp8h02", "chain_params": { "PARAM1": "'$PARAM1'", "PARAM2": "'$PARAM2'" } }'

and the chained event (event2) is just
#!/usr/bin/env bash
echo $PARAM1
echo $PARAM2

But when I run event1 I get this in the log, but the chained event doesn't get run

# Job ID: jlo3n9tzw04
# Event Title: Chain event 1
# Hostname: apchop01.example.com
# Date/Time: 2023/10/24 12:25:10 (GMT+11)

FOO
BAR
{"chain":"elo3mvp8h02","chain_params":{"PARAM1":"FOO","PARAM2":"BAR"}}

# Job completed successfully at 2023/10/24 12:25:10 (GMT+11).
# End of log.

**Comment by jhuckaby** [comment]
You gotta check this box for your Shell Plugin events:

<img width="366" alt="Screenshot 2023-10-23 at 6 42 41 PM" src="https://github.com/jhuckaby/Cronicle/assets/700765/1610c376-c91d-4a11-9230-2bb14df701d3">

**Comment by kranskydog** [comment]
Ahh

**Comment by kranskydog** [comment]
Perfect. Thanks.
